test_resource/test_repo/src/test_guardrails.py

```diff
@@ -0,0 +1,531 @@
+        import os
+        ini_path = os.path.abspath(os.path.join
+                                   (os.path.dirname(__file__), os.pardir))
+        ini_path = os.path.join(ini_path, "test_resource", "guardrail.ini")
+        config = ConfigParser()
+        config.read(ini_path)
+        self.report_folder = (config.get('folder', 'report_folder'))
+        if not os.path.exists(self.report_folder):
+            os.makedirs(self.report_folder)
+
+    def tearDown(self):
+        """"Deletes the log files created."""
+        import os
+        ini_path = os.path.abspath(os.path.join
+                                   (os.path.dirname(__file__), os.pardir))
+        file_name = os.path.join(ini_path, "guardrails", "guardrails.log")
+        if os.path.exists(file_name):
+            open(file_name, 'w').close()
+
+    @staticmethod
+    def stub_globals(guardails_obj):
+        """Function to stub all globals"""
+        guardails_obj.src_folder = "abc"
+        guardails_obj.test_folder = "test"
+        guardails_obj.pytest = "abc"
+        guardails_obj.report_folder = "abc/report"
+        guardails_obj.cyclo_exclude = "abc/None"
+        guardails_obj.python = "python"
+        guardails_obj.pylintrc = ".pylintrc"
+        guardails_obj.covrc = ".cov"
+        guardails_obj.dup_token = 10
+        guardails_obj.percent_cov = 80
+        guardails_obj.allow_dup = 20
+        guardails_obj.cc_limit = 10
+        guardails_obj.allow_mutants = 10
+        guardails_obj.all_folders = guardails_obj.src_folder + " " + guardails_obj.test_folder
+        guardails_obj.linting = True
+        guardails_obj.cpd = True
+        guardails_obj.cov = True
+        guardails_obj.mutation = True
+        guardails_obj.deadcode = True
+        guardails_obj.cycloc = True
+        guardails_obj.lint_ignore = "x"
+        guardails_obj.programming_language = "java"
+        guardails_obj.jscpd_ignore = "y"
+        guardails_obj.dead_code_ignore = "z"
+
+    def generate_pylint_cmd(self):  # pylint: disable=R0201
+        """Function stubbed to return generate_pylint_cmd method return value"""
+        return "xyz"
+
+    def stub_validate_return(self, val, message, guardrail):  # pylint: disable=W0613,R0201
+        """Function stubbed to return false"""
+        return False
+
+    def return_jscpd_false(self, val, message):  # pylint: disable=W0613,R0201
+        """Function stubbed to set parse_jscpd_report_json method value to false"""
+        return False
+
+    @staticmethod
+    def stub_get_all_func_cnn():
+        """Function stubbed to return get_all_func_cnn function return value"""
+        list_data = {
+            'get_file_names(...) at C:\\Projects\\PythonRepo\\python_sample\\FunctionDefExtractor'
+            '\\functiondefextractor\\core_extractor.py:11': '3',
+            'get_function_names(...) at C:\\Projects\\PythonRepo\\python_sample\\FunctionDefExtractor'
+            '\\functiondefextractor\\core_extractor.py:28': '7'}
```

Comment:
Though the files are used for data frame generation check, Suggested to remove the C:\"hard coded paths" Comment applicable across test resource files